src/presentational/Classroom.jsx

```diff
@@ -23,15 +24,43 @@ export default class Classroom extends Component {
   }
 
   componentWillReceiveProps(nextProps){
-    let id = nextProps.data.id;
-    let token = nextProps.data.attributes.join_token;
+    let classrooms = nextProps.data;
     this.state = {
-      url: routes.root + routes.students + 'join?id=' + id + '&token=' + token
+      url: routes.root + routes.students + 'join?id=' + classrooms.id + '&token=' + classrooms.attributes.join_token
     };
   }
 
+  renderStudentList(allMembers, classroomMembers) {
+    const list = (allMembers.length > 0) ? allMembers : 'No student found';
```

Comment:
This sets `list` to a string if there are no students, which will break any code that depends on it being an array. MAybe set it `[]` instead?